src/encoder.rs

```diff
@@ -3449,9 +3419,37 @@ pub fn encode_show_existing_frame<T: Pixel>(
   fi: &FrameInvariants<T>, fs: &mut FrameState<T>, inter_cfg: &InterConfig,
 ) -> Vec<u8> {
   debug_assert!(fi.show_existing_frame);
+  let obu_extension = 0;
+
   let mut packet = Vec::new();
 
-  write_obus(&mut packet, fi, fs, inter_cfg).unwrap();
+  if fi.frame_type == FrameType::KEY {
+    write_key_frame_obus(&mut packet, fi, obu_extension).unwrap();
+  }
+  let mut buf1 = Vec::new();
+  let mut buf2 = Vec::new();
+  {
+    let mut bw2 = BitWriter::endian(&mut buf2, BigEndian);
+    bw2.write_frame_header_obu(fi, fs, inter_cfg).unwrap();
+  }
+
+  {
+    let mut bw1 = BitWriter::endian(&mut buf1, BigEndian);
+    bw1.write_obu_header(ObuType::OBU_FRAME_HEADER, obu_extension).unwrap();
+  }
+  packet.write_all(&buf1).unwrap();
+  buf1.clear();
+
+  {
+    let mut bw1 = BitWriter::endian(&mut buf1, BigEndian);
```

Comment:
Here there are two consecutive uses of `bw1`, they can probably be merged.

Reply:
It's the same as before this patch, and i can't seem to find a way to merge them.

Reply:
Okay, not important for this patch then, I will look into it on my own.